### PR TITLE
HTTP Client and Server

### PR DESCRIPTION
**Pull request by jdk514**
Team HW1 - Katie, Lucas, and Joel. Simple http server and client. Solves issues #14 and #15.

### CODE REVIEW

File: sockets/.gitignore [hunk 1]
```diff
@@ -1,2 +1,3 @@
 server-tcp
 client-tcp
+server-http
```
Comment (twood02) [inline sockets/.gitignore:3]: might as well add `web-get` while you are at it.

File: sockets/server-http.c [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+#include <sys/socket.h>
+#include <netinet/in.h>
+#include <netdb.h>
+#include <unistd.h>
+#include <inttypes.h>
+#include <string.h>
+#include <sys/stat.h>
+
+/****************************************
+ Author: Tim Wood
```
Comment (twood02) [inline sockets/server-http.c]: put your names before mine in author list

File: sockets/server-http.c [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+typedef struct http_response{
+    char* version;
+    char* response_code;
+    char* host;
+    char* content_type;
+    int len;
+    char* body;
+} http_response;
+
+http_response *http_response_alloc() {
```
Comment (twood02) [inline sockets/server-http.c]: Style Nazi: braces for functions go on new lines

File: sockets/webget.c [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+	//bytes received from server
+	int bytes_received = -1;
+	//data received from http request--will be printed out to user
+	char * recv_data = (char *)malloc(sizeof(char*)*1000);
+	int o;
+
+        /* Command line args:
+                -p port
+                -h host name or IP
+		-i http request
```
Comment (twood02) [inline sockets/webget.c]: Why do you need both hostname and server IP?  You should be able to figure out one from the other after calling getaddrinfo().  (also this explanation of -i doesn't make sense to me)

**Comment by twood02** [comment]
overall looks great!  I've added a few comments you should fix.

**Comment by kstats** [comment]
Code has been revised to reflect your suggestions.

**Comment by twood02** [comment]
looks good!

**Pull request merged**